spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -615,32 +540,241 @@ def copy(self):
+
+    def data(self, index, role):
+        """
+        Get the data for the header.
+
+        This is used when a header has levels.
+        """
+        if not index.isValid() or \
+           index.row() >= self._shape[0] or \
+           index.column() >= self._shape[1]:
+            return None
+        row, col = ((index.row(), index.column()) if self.axis == 0
+                    else (index.column(), index.row()))
+        if role == Qt.BackgroundRole:
+            prev = self.model.header(self.axis, col - 1, row) if col else None
+            cur = self.model.header(self.axis, col, row)
+            return self._palette.midlight() if prev != cur else None
+        if role != Qt.DisplayRole:
+            return None
+        if self.axis == 0 and self._shape[0] <= 1:
+            return None
+        return to_text_string(self.model.header(self.axis, col, row))
+
+
+class DataFrameLevelModel(QAbstractTableModel):
+    """
+    Data Frame level class.
+
+    This class is used to represent index levels in the DataFrameEditor. When
+    using MultiIndex, this model creates labels for the index/header as Index i
+    for each section in the index/header
+
+    Based on the gtabview project (Level4ExtModel).
+    For more information please see:
+    https://github.com/wavexx/gtabview/blob/master/gtabview/viewer.py
+    """
+
+    def __init__(self, model, palette, font):
+        super(DataFrameLevelModel, self).__init__()
+        self.model = model
+        self._background = palette.dark().color()
+        if self._background.lightness() > 127:
+            self._foreground = palette.text()
+        else:
+            self._foreground = palette.highlightedText()
+        self._palette = palette
+        font.setBold(True)
+        self._font = font
+
+    def rowCount(self, index=None):
+        """Get number of rows (number of levels for the header)."""
+        return max(1, self.model.header_shape[0])
+
+    def columnCount(self, index=None):
+        """Get the number of columns (number of levels for the index)."""
+        return max(1, self.model.header_shape[1])
+
+    def headerData(self, section, orientation, role):
+        """
+        Get the text to put in the header of the levels of the indexes.
+
+        By default it returns 'Index i', where i is the section in the index
+        """
+        if role == Qt.TextAlignmentRole:
+            if orientation == Qt.Horizontal:
+                return Qt.AlignCenter | Qt.AlignBottom
+            else:
+                return Qt.AlignRight | Qt.AlignVCenter
+        if role != Qt.DisplayRole and role != Qt.ToolTipRole:
+            return None
+        if self.model.header_shape[0] <= 1 and orientation == Qt.Horizontal:
+            return _('Index')
+        elif self.model.header_shape[0] <= 1:
+            return None
+        elif self.model.header_shape[1] <= 1 and orientation == Qt.Vertical:
+            return None
+        return _('Index ') + to_text_string(section)

```

Comment:
Write this as `_('Index') + ' ' + ...` so that translators do not have to translate both `'Index'` (without space) and `'Index '` (with a trailing space).